Fix configuration with CMake >= 4:

Compatibility with CMake < 3.5 has been removed from CMake.
Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
to tell CMake that the project requires at least <min> but has been updated
to work with policies introduced by <max> or earlier.
Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.

The fix for CMakeLists.txt and ExampleDriver/CMakeLists.txt is from:

https://github.com/pothosware/SoapySDR/commit/fbf9f3c328868f46029284716df49095ab7b99a6

python/CMakeLists.txt is not yet fixed upstream; see:

https://github.com/pothosware/SoapySDR/issues/360
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -1,7 +1,7 @@
 ########################################################################
 # Project setup
 ########################################################################
-cmake_minimum_required(VERSION 3.1.0)
+cmake_minimum_required(VERSION 3.3...3.10)
 project(SoapySDR)
 enable_language(CXX)
 enable_testing()
--- ExampleDriver/CMakeLists.txt.orig
+++ ExampleDriver/CMakeLists.txt
@@ -2,7 +2,7 @@
 # Project setup -- only needed if device support is a stand-alone build
 # We recommend that the support module be built in-tree with the driver.
 ########################################################################
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 2.6...3.10)
 project(SoapySDRMyDevice CXX)
 enable_testing()
 
--- python/CMakeLists.txt.orig
+++ python/CMakeLists.txt
@@ -1,7 +1,7 @@
 ########################################################################
 # Project setup
 ########################################################################
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 2.8...3.10)
 project(SoapySDRPython CXX)
 enable_testing()
 
